matminer/featurizers/site.py

```diff
@@ -464,14 +472,31 @@ def featurize(self, struct, idx):
 
         cn_fingerprint = []
 
+        if self.chem_info is not None:
+            absdeltas = {}
+            for prop in self.chem_props:
+                absdeltas[prop] = 0
+            Nwt = 0
+            elem_cent = struct.sites[idx].species_string
+
         for k in range(max_cn):
             cn = k + 1
             wt = nndata.cn_weights.get(cn, 0)
-            if cn not in self.ops:
-                cn_fingerprint.append(0)
-            else:
+            if cn in self.ops:
                 for op in self.ops[cn]:
-                    if op == "wt" or wt == 0:
+                    if op == "wt":
+                        cn_fingerprint.append(wt)
+                        # Compute additional chemistry-related features
+                        if self.chem_info is not None and wt != 0:
+                            Nwt += wt
+                            for prop in self.chem_props:
+                                delem = self.chem_info[prop]
+                                tmp = 0
```

Comment:
generally this kind of ``tmp`` variable is discouraged. You'd instead do something like

```
my_var = sum([delem[d["site"].species_string] - delem[elem_cent] for d in nndata.cn_nninfo[cn]])
```